internal/client.go

```diff
@@ -431,10 +431,16 @@ type (
 		WorkflowTaskTimeout time.Duration
 
 		// WorkflowIDReusePolicy - Whether server allow reuse of workflow ID, can be useful
-		// for dedup logic if set to RejectDuplicate.
+		// for dedupe logic if set to RejectDuplicate.
 		// Optional: defaulted to AllowDuplicate.
 		WorkflowIDReusePolicy enumspb.WorkflowIdReusePolicy
 
+		// When WorkflowExecutionErrorWhenAlreadyStarted is true, Client.ExecuteWorkflow will return an error if the
+		// workflow id has already been used and WorkflowIDReusePolicy would disallow a re-run.
```

Comment:
Document what happens when it is set to false.

Reply:
When WorkflowExecutionErrorWhenAlreadyStarted  is true then WorkflowExecutionErrorWhenAlreadyStarted will be always thrown for a workflow that is open independently of WorkflowIdReusePolicy. And for the closed workflows it is going to throw if no duplicated workflow can be created according to the WorkflowIdReusePolicy.

When WorkflowExecutionErrorWhenAlreadyStarted is false then WorkflowExecutionErrorWhenAlreadyStarted is thrown for open workflows only for WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE and never for the closed ones.